leshan-core/src/test/java/org/eclipse/leshan/core/VersionTest.java

```diff
@@ -35,9 +39,66 @@ public void is_supported_tests() {
     @Test
     public void compare_tests() {
         assertTrue(new Version("1.0").compareTo(new Version("1.2")) < 0);
+        assertTrue(new Version("1.0").olderThan(new Version("1.2")));
+        assertNotEquals(new Version("1.0").hashCode(), new Version("1.2").hashCode());
```

Comment:
It was intended, so that the `hashCode()` generation is also verified. But I agree this became quite chaotic… I just cleaned that up (and also fixed the formatting issues hopefully :slightly_smiling_face:)

Reply:
If you want to check `equals`/`hashcode`, probably better to use `EqualsVerifier`. 

Replacing you code by  : 
```java
@Test
public void assertEqualsHashcode() {
    // TODO we should not use EqualsVerifier.simple()
    // but implement a right hashcode/equals way
    // see : https://github.com/eclipse-leshan/leshan/issues/1504
    EqualsVerifier.simple().forClass(Version.class).verify();
}
```
(If you are curious about that you could look at : #1504)

I will do this last little change and then integrate this PR in `master`.